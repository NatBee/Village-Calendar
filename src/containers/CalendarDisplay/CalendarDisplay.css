.calendar-display {
  padding-bottom: 120px;
  text-align: -webkit-center;
  font-family: 'Exo 2', sans-serif;
  margin: 75px 0; }
