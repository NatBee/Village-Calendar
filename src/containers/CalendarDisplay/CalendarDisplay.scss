* {
  margin: 0;
  padding: 0;
}

.calendar-display {
  padding-bottom: 120px;
  text-align: -webkit-center;
  font-family: 'Exo 2', sans-serif;
  margin: 50px 0;
  position: fixed;
  top: 10%;
  width: 100%;
} 