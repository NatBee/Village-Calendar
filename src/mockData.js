export const mockData = {
  user: {
    user: 'Dk', 
    credential: 'Xg', 
    additionalUserInfo: 'Pg', 
    operationType: "signIn"
  },
  events: [
    {
      start: "2018-02-09T09:00:00-07:00",
      end: "2018-02-09T10:00:00-07:00",
      title: "Super Fun Party",
      description: "Fun with friends",
      eventID: "7js6e24fo0kin4qcotoegu37hg",
      location: 'Mi Casa'
    }
  ],
  token: 'token_number',
  calendarID: 'calendarID',
  village: [{natalie: 'nataliesmontoya@gmail.com'}],
  time: {
    startTime: "2018-02-28T19:00:00.000Z", 
    endTime: "2018-02-28T20:00:00.000Z"
  },
  event: {
    start: "2018-02-09T09:00:00-07:00",
    end: "2018-02-09T10:00:00-07:00",
    title: "Super Fun Party",
    description: "Fun with friends",
    eventID: "7js6e24fo0kin4qcotoegu37hg",
    location: 'Mi Casa'
  }
}