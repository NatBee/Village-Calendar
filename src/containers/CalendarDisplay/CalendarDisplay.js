import React, { Component } from 'react';
import moment from 'moment';
import BigCalendar from 'react-big-calendar';
import './CalendarDisplay.css';
import 'react-big-calendar/lib/css/react-big-calendar.css';
import { getUpcomingEvents } from '../../helper/apiCall';
import { connect } from 'react-redux';
import { loadUpcomingEvents, setTimeAddEvent, setEvent, removeEvent } from '../../actions/index';
import EventCard from '../EventCard/EventCard';
import EditEventCard from '../EditEventCard/EditEventCard';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

BigCalendar.momentLocalizer(moment);

class CalendarDisplay extends Component {
  constructor(props) {
    super(props);

    this.state = {
      eventCard: 'calendar'
    }
  }

  async componentDidMount() {
    const events = await getUpcomingEvents(this.props.calendarID); 
    this.props.loadUpcomingEvents(events);
  }

  editEvent = (event) => {
    this.props.history.push('/editevent')
    this.props.setEvent(event);
    this.setState({eventCard: 'edit'})
  }

  setEventTime = async (start, end) => {
    const time = {  
      startTime: start,
      endTime: end    
    };
    this.props.setTimeAddEvent(time);
    this.setState({eventCard: 'add'})
    await this.addEventCard();
  }

  addEventCard = () => {
    this.props.history.push('/addevent')
  }

  display = () => {
    if(this.state.eventCard === 'calendar') {
      return (
        <BigCalendar 
          style={{height: '420px'}} 
          events={this.props.events} 
          selectable
          views={{
              month: true,
              day: true
          }}
          defaultDate={new Date(Date.now())}
          onSelectEvent={event => this.editEvent(event)}
          onSelectSlot={slotInfo => this.setEventTime(slotInfo.start.toISOString(), slotInfo.end.toISOString())}
        />
      )
    } else if(this.state.eventCard === 'add') {
      return (
        <EventCard/>
      )
    } else {
      return (
        <EditEventCard/>
      )
    }
  }

  render() {
    return (
      <div>
        { this.display() }
      </div>
    )
  }

}

export const mapStateToProps = (store) => ({
  events: store.events,
  time: store.time,
  calendarID: store.calendarID,
  event: store.event,
  token: store.token
})

export const mapDispatchToProps = (dispatch) => ({
  loadUpcomingEvents: (events) => dispatch(loadUpcomingEvents(events)),
  setTimeAddEvent: (time) => dispatch(setTimeAddEvent(time)),
  setEvent: (event) => dispatch(setEvent(event)),
  removeEvent: (event) => dispatch(removeEvent(event))
})
 
CalendarDisplay.propTypes = {
  events: PropTypes.array,
  time: PropTypes.object,
  calendarID: PropTypes.string,
  event: PropTypes.object,
  token: PropTypes.string,
  loadUpcomingEvents: PropTypes.func,
  setTimeAddEvent: PropTypes.func,
  setEvent: PropTypes.func,
  removeEvent: PropTypes.func,
  getUpcomingEvents: PropTypes.func,
  editEvent: PropTypes.func,
  setTimeEvent: PropTypes.func,
  addEventCard: PropTypes.func,
  display: PropTypes.func,
  history: PropTypes.object,
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(CalendarDisplay));