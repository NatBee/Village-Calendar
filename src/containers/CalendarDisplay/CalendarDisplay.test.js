/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { CalendarDisplay, mapStateToProps, mapDispatchToProps } from './CalendarDisplay';
import { mockData } from '../../mockData';

global.localStorage = {
  getItem(keyword) {
    if (!global.localStorage[keyword]) {
      return null;
    }
    return JSON.stringify(global.localStorage[keyword]);
  },
  setItem(keyword, value) {
    global.localStorage[keyword] = value;
  }
};

describe('CalendarDisplay', () => {
  let wrapper;
  const DATE_TO_USE = new Date('2016');
  const _Date = Date;
  global.Date = jest.fn(() => DATE_TO_USE);
  global.Date.UTC = _Date.UTC;
  global.Date.parse = _Date.parse;
  global.Date.now = _Date.now;

  beforeEach(() => {
    localStorage.setItem('ouath2-access-token', mockData.token);
    wrapper = shallow(<CalendarDisplay events={ mockData.events } />);
  })

  it('should match snapshot', () => {
    
    expect(wrapper).toMatchSnapshot();
  })

  it('should map to the store correctly', () => {
    const mockStore = {
      events: mockData.events,
      time: mockData.time,
      calendarID: mockData.calendarID,
      event: mockData.event,
      token: mockData.token
    }
    const mapped = mapStateToProps(mockStore);
    expect(mapped).toEqual(mockStore);
  })

  it('should call the dispatch function when using a loadUpcomingEvents from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.loadUpcomingEvents();
    expect(mockDispatch).toHaveBeenCalled();
  })

  it('should call the dispatch function when using setTimeAddEvent function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.setTimeAddEvent();
    expect(mockDispatch).toHaveBeenCalled();
  })  

  it('should call the dispatch function when using setEvent function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.setEvent();
    expect(mockDispatch).toHaveBeenCalled();
  })  

  it('should call the dispatch function when using removeEvent function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.removeEvent();
    expect(mockDispatch).toHaveBeenCalled();
  })

  it('should have a default state', () => {
    const expected = {
      eventCard: 'calendar'
    }
    expect(wrapper.state()).toEqual(expected);
  })
})