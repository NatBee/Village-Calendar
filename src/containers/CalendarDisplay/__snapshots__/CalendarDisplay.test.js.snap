// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDisplay should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CalendarDisplay
    events={
      Array [
        Object {
          "description": "Fun with friends",
          "end": "2018-02-09T10:00:00-07:00",
          "eventID": "7js6e24fo0kin4qcotoegu37hg",
          "location": "Mi Casa",
          "start": "2018-02-09T09:00:00-07:00",
          "title": "Super Fun Party",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Uncontrolled(Calendar)
        defaultDate={2018-02-26T03:47:05.791Z}
        events={
          Array [
            Object {
              "description": "Fun with friends",
              "end": "2018-02-09T10:00:00-07:00",
              "eventID": "7js6e24fo0kin4qcotoegu37hg",
              "location": "Mi Casa",
              "start": "2018-02-09T09:00:00-07:00",
              "title": "Super Fun Party",
            },
          ]
        }
        onSelectEvent={[Function]}
        onSelectSlot={[Function]}
        selectable={true}
        style={
          Object {
            "height": "420px",
          }
        }
        views={
          Object {
            "day": true,
            "month": true,
          }
        }
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "defaultDate": 2018-02-26T03:47:05.791Z,
        "events": Array [
          Object {
            "description": "Fun with friends",
            "end": "2018-02-09T10:00:00-07:00",
            "eventID": "7js6e24fo0kin4qcotoegu37hg",
            "location": "Mi Casa",
            "start": "2018-02-09T09:00:00-07:00",
            "title": "Super Fun Party",
          },
        ],
        "onSelectEvent": [Function],
        "onSelectSlot": [Function],
        "selectable": true,
        "style": Object {
          "height": "420px",
        },
        "views": Object {
          "day": true,
          "month": true,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Uncontrolled(Calendar)
          defaultDate={2018-02-26T03:47:05.791Z}
          events={
            Array [
              Object {
                "description": "Fun with friends",
                "end": "2018-02-09T10:00:00-07:00",
                "eventID": "7js6e24fo0kin4qcotoegu37hg",
                "location": "Mi Casa",
                "start": "2018-02-09T09:00:00-07:00",
                "title": "Super Fun Party",
              },
            ]
          }
          onSelectEvent={[Function]}
          onSelectSlot={[Function]}
          selectable={true}
          style={
            Object {
              "height": "420px",
            }
          }
          views={
            Object {
              "day": true,
              "month": true,
            }
          }
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultDate": 2018-02-26T03:47:05.791Z,
          "events": Array [
            Object {
              "description": "Fun with friends",
              "end": "2018-02-09T10:00:00-07:00",
              "eventID": "7js6e24fo0kin4qcotoegu37hg",
              "location": "Mi Casa",
              "start": "2018-02-09T09:00:00-07:00",
              "title": "Super Fun Party",
            },
          ],
          "onSelectEvent": [Function],
          "onSelectSlot": [Function],
          "selectable": true,
          "style": Object {
            "height": "420px",
          },
          "views": Object {
            "day": true,
            "month": true,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
